### ISSUE TITLE
Build error in wxrc

### ISSUE THREAD

**Issue by pietvo** [wxWidgets/wxWidgets#23359]
### Description

I discovered this problem when installing wxPython on MacOS with macports.

`wxrc` is built with wrong references to the wxWidgets dynamic libraries.

Here's what happens when you run it:

```
wxrc-3.10
dyld[92389]: Library not loaded: /opt/local/var/macports/build/_Users_pieter_Projects_MACPORTS_python_py-wxpython-4.2/py310-wxpython-4.2/work/wxPython-4.2.0/build/wxbld/lib/libwx_baseu_xml-3.2.0.0.0.dylib
  Referenced from: <D3C3E569-36F8-3081-90A4-B59A4C4ED3E7> /opt/local/Library/Frameworks/Python.framework/Versions/3.10/bin/wxrc-3.2
  Reason: tried: '/opt/local/var/macports/build/_Users_pieter_Projects_MACPORTS_python_py-wxpython-4.2/py310-wxpython-4.2/work/wxPython-4.2.0/build/wxbld/lib/libwx_baseu_xml-3.2.0.0.0.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/local/var/macports/build/_Users_pieter_Projects_MACPORTS_python_py-wxpython-4.2/py310-wxpython-4.2/work/wxPython-4.2.0/build/wxbld/lib/libwx_baseu_xml-3.2.0.0.0.dylib' (no such file), '/opt/local/var/macports/build/_Users_pieter_Projects_MACPORTS_python_py-wxpython-4.2/py310-wxpython-4.2/work/wxPython-4.2.0/build/wxbld/lib/libwx_baseu_xml-3.2.0.0.0.dylib' (no such file), '/usr/local/lib/libwx_baseu_xml-3.2.0.0.0.dylib' (no such file), '/usr/lib/libwx_baseu_xml-3.2.0.0.0.dylib' (no such file, not in dyld cache)
Abort trap: 6
```

**Diagnosis**: `configure` generates a shell script `change-install-names` that fixes the references to the `wx` libraries with `install_name_tool`. This script is run at the end of the installation procedure. It also writes a command to this file to fixup the library references in `wxrc-${WX_RELEASE}`. In a previous version this worked but in the current version the last command is written to `change-install-names` first, and therefore it will be overwritten by the main script.

 The **solution** is to move the writing of the last command to a place after the first one. Here is my suggestion: I have patched `configure` but it should (also) go to `configure.in`.

```
--- configure.~1~	2022-07-06 00:27:55.000000000 +0200
+++ configure	2023-03-17 15:00:27.498675068 +0100
@@ -27350,16 +27350,6 @@
 
 fi
 
-if test "$wxUSE_XML" != "yes"; then
-    { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: XML library not built, cannot build wxrc" >&5
-$as_echo "$as_me: WARNING: XML library not built, cannot build wxrc" >&2;}
-    USE_XML=0
-else
-    USE_XML=1
-    echo "${HOST_PREFIX}install_name_tool \${changes} \${2}/wxrc-${WX_RELEASE}" >> change-install-names
-fi
-
-
 
 if test "$wxUSE_LIBMSPACK" != "no"; then
     ac_fn_c_check_header_compile "$LINENO" "mspack.h" "ac_cv_header_mspack_h" "
@@ -41513,6 +41503,16 @@
     SAMPLES_SUBTREES="$SAMPLES_SUBTREES html"
 fi
 
+if test "$wxUSE_XML" != "yes"; then
+    { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: XML library not built, cannot build wxrc" >&5
+$as_echo "$as_me: WARNING: XML library not built, cannot build wxrc" >&2;}
+    USE_XML=0
+else
+    USE_XML=1
+    echo "${HOST_PREFIX}install_name_tool \${changes} \${2}/wxrc-${WX_RELEASE}" >> change-install-names
+fi
+
+
 USE_XRC=0
 if test "$wxUSE_XRC" = "yes"; then
     if test "$wxUSE_XML" != "yes"; then


```

### Platform and version information

- wxWidgets version  you are building: 3.2.2
- wxWidgets port  you are building: wxOSX
- OS  and its version: MacOS 13.2.1 (Ventura)
- Compiler  being used: Apple clang version 14.0.0 (clang-1400.0.29.202)

**Comment by vadz** [comment]
This seems to be the same as #22620, are you sure you have 3.2.2 version of configure? It shouldn't contain the lines removed by your patch.

**Comment by pietvo** [comment]
I have the version that is included in wxPython 4.2.0. I thought that it was 3.2.2, but now I look again, I see in the repository that the version was updated to 3.2.2 in master a month ago, so that's after the wxPython 4.2.0 release. Sorry about that.

**Comment by vadz** [comment]
Then I think it's really just the same problem and it was already fixed in both branches.
